### PR TITLE
Improve documention troubleshooting missing linker. Fix #32208.

### LINKED ISSUE #32208: Rust requires an installed C compiler for linking, but book never mentions this is necessary
From what I understand, Rust executes `cc` for linking programs. However, the book doesn't mention that a C compiler needs to be installed before installing Rust, resulting in a confusing error along these lines when one attempts to compile a Rust program:
```
error: could not exec the linker `cc`: No such file or directory (os error 2)
```

### PR DESCRIPTION
**Pull request by pmatos**
@steveklabnik is this in the direction on how you want to see #32208 fixed?

### CODE REVIEW

**Comment by steveklabnik** [comment]
@pmatos yes, looks great to me, thank you!

@bors: r+ rollup

File: src/doc/book/getting-started.md
Comment (retep998) [inline src/doc/book/getting-started.md]: `link.exe`. Also it doesn't need to be in `PATH` because rustc can find msvc itself.

**Comment by steveklabnik** [comment]
@bors: r-

gah, read that one wrong, sorry

**Comment by sfackler** [comment]
@bors r-

Pending @retep998's link.exe comment.

**Comment by pmatos** [comment]
@retep998 thanks, preparing new pull request.

**Comment by pmatos** [comment]
@retep998 I tried with the new pull request to take your comments into consideration but if there's something else that can be improved, please let me know.

File: src/doc/book/getting-started.md
Comment (steveklabnik) [inline src/doc/book/getting-started.md]: Windows, not windows.

**Review by steveklabnik — CHANGES_REQUESTED** [review]
One tiny nit, and then if @retep998 is happy, I'm happy.

File: src/doc/book/getting-started.md
Comment (retep998) [inline src/doc/book/getting-started.md]: Microsoft Visual ___C++___ Build Tools

**Comment by pmatos** [comment]
@steveklabnik Done. @retep998 Done. 
The last two commits should have fixed the concerns you had. Thanks for the help on getting this pull request complete.

**Comment by steveklabnik** [comment]
Great! Thanks so much for your patience here too.

One last question before I merge: it seems that the commit authorship is different than your github account. Is that what you intended? If so, it's fine, but otherwise, you might want to fix that before we merge :smile:

**Review by steveklabnik — APPROVED** [review]

**Comment by pmatos** [comment]
You're right. Let me fix authorship. Sorry for this.

**Comment by steveklabnik** [comment]
It's no worries! :)

**Comment by pmatos** [comment]
@steveklabnik I have rewritten history on my end in order to change commit authorship which worked but now github falsely shows branch conflicts. Would you be ok with accepting this or do you want me to restart a brand new pull request on this?

**Comment by steveklabnik** [comment]
Re-writing the history is totally fine :)

**Comment by steveklabnik** [comment]
(the message in the bot is meant to be talking about _during review_, not administrative stuff like this at the end :smile: )

**Comment by pmatos** [comment]
Great! So we are good to go. Happy to contribute. Anything else you have in mind that I could tackle next? 

On 27 September 2016 21:27:49 CEST, Steve Klabnik <notifications@github.com> wrote:
>(the message in the bot is meant to be talking about _during review_,
>not administrative stuff like this at the end :smile: )
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/rust-lang/rust/pull/36672#issuecomment-249971613

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by steveklabnik** [comment]
(You have to force push, it didn't actually update the PR)

Thanks! We have the `A-docs` tag for doc stuff, and not all of them are super difficult: https://github.com/rust-lang/rust/issues?q=is%3Aopen+is%3Aissue+label%3AA-docs

https://github.com/rust-lang/rust/issues?q=is%3Aopen+is%3Aissue+label%3AA-docs+label%3AE-easy are specifically the ones tagged easy. Lots to do!

**Comment by steveklabnik** [comment]
Oh, it's not that you didn't force push, it's that when you did, it somehow got on the wrong base. Try to rebase things and force push again.

**Comment by pmatos** [comment]
@steveklabnik Is this ok now?

**Comment by steveklabnik** [comment]
Hm, it still says 250+ commits :eyes: 

If you want some help, I can do it: I'd just need you to follow the second half here: https://github.com/blog/2247-improving-collaboration-with-forks

**Comment by pmatos** [comment]
@steveklabnik The checkbox to allow edits from maintainers is enabled. Thanks for your help. I have no idea at the moment what kind of weird games git/github is playing with me.

**Comment by steveklabnik** [comment]
Whew! @pmatos , I don't know what happened, but something was very wrong. Trying to deal with this made my git go a bit... intense.

2016-09-27-163436_706x57_scrot

However, I managed to make it work. Here's what I did:

1. `git rebase -i` to squash the four commits into one.
2. `git remote add upstream https://github.com/rust-lang/rust/`
3. `git fetch upstream`
4. `git reset --hard upstream/master`
5. `git cherry-pick cb3b03c`
6. `git push origin master -f`

Specifically step 4 says "make this branch equal to master" and then the cherry-pick lets you apply a random commit on top. Whew!

Can you verify that this commit is what you wanted to send in? It looks okay with me, but since I just re-wrote your history, it's important that you read it before we merge, since it's in your name :smile:

**Comment by pmatos** [comment]
@steveklabnik many thanks for your time. roll it up. :) see in the next pull request.

**Comment by steveklabnik** [comment]
Great! I think one of the problems might have been that it was on `master` instead of a branch, so maybe make a branch per PR next time? :smile:

**Comment by pmatos** [comment]
@steveklabnik I will use a branch per PR next time. Thanks.

**Pull request merged**
